core/src/main/java/feign/Headers.java

```diff
@@ -24,6 +24,7 @@
  * }) void post(&#64;Param("token") String token);
  * ...
  * </pre>
+ * <br> Note that if you'd like curly braces literally in the header, urlencode them first.
```

Comment:
Nit: maybe strong this to be consistent with the following line, or perhaps bullet the notes